.github/workflows/build.yaml

```
@@ -0,0 +1,18 @@
+---
+on: [push, pull_request]
+
+jobs:
+  compile:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+      - run: npm install
+      - run: npm run build
+      - run: npm run format
+      - uses: fregante/setup-git-user@v1
+      - if: github.event_name == 'push' && github.ref_name == 'main'
+        run: |
+          git add .
+          git commit -m "[ci skip] compile and lint $GITHUB_SHA"
+          git push
```

```suggestion
          git push origin main
```

Need to be explicit here as remote tracking of branch not setup